File: bootcamp-app/src/aggregate/sdo/CommentCdo.ts
import {IdName, newIdName} from "../vo/IdName";

export interface CommentCdo {
    user: IdName;
    content: string;
    upperBoardId: string;
    upperCommentId: string;
}

export const newCommentCdo = (user = newIdName(), content='', upperBoardId='', upperCommentId=''): CommentCdo => {
    return {
        user, content, upperBoardId, upperCommentId
    }
}

File: bootcamp/src/main/java/io/nextree/board/aggregate/sdo/CommentCdo.java
package io.nextree.board.aggregate.sdo;

import io.nextree.Util.IdName;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CommentCdo {
    //
    transient private String id; // front에서 BoardCreate객체를 넘겨줄때 id는 비워져서 올거라는 알림
    private IdName user;
    private String content;
    private String upperBoardId;
    private String upperCommentId;
}
